### ISSUE TITLE
`no-curly-component-invocation` false-negatives with yielded "templates"

### ISSUE THREAD

**Issue by boris-petrov** [ember-template-lint/ember-template-lint#3275]
```hbs
<MyYieldingComponent as |foo|>
  {{#foo.bar}}
  {{/foo.bar}}
</MyYieldingComponent>
```

This doesn't trigger a `no-curly-component-invocation` error but I believe it should - for consistency, the yielded "templates" should also be using angle-brackets.

**Comment by boris-petrov** [comment]
My bad, this is working totally fine.
